vault/resource_consul_secret_backend_role.go

```diff
@@ -138,21 +141,23 @@ func consulSecretBackendRoleGetBackend(d *schema.ResourceData) string {
 	}
 }
 
-func consulSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
-	client, e := provider.GetClient(d, meta)
-	if e != nil {
-		return e
+func consulSecretBackendRoleWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
 	}
 
 	name := d.Get("name").(string)
 
 	backend := consulSecretBackendRoleGetBackend(d)
 	if backend == "" {
-		return fmt.Errorf("No backend specified for Consul secret backend role %s", name)
+		return diag.Errorf("No backend specified for Consul secret backend role %s", name)
 	}
 
 	path := consulSecretBackendRolePath(backend, name)
 
+	// This loads either the consul_policies or policies field, depending on which the
+	// user provided, and then stores it under both keys in the data.
```

Comment:
Is this comment still valid? Looks like we are no longer storing the `policies` variable under both keys, and are conditionally only setting one of them